weavelet.go

```diff
@@ -118,15 +112,42 @@ func newWeavelet(ctx context.Context, componentInfos []*codegen.Registration) (*
 			// Discard all log entries.
 			logger: slog.New(slog.HandlerOptions{Level: slog.LevelError + 1}.NewTextHandler(os.Stdout)),
 		}
-		byName[info.Name] = c
-		byType[info.Iface] = c
+		w.componentsByName[info.Name] = c
+		w.componentsByType[info.Iface] = c
 	}
-	main, ok := byName["main"]
+	main, ok := w.componentsByName["main"]
 	if !ok {
 		return nil, fmt.Errorf("internal error: no main component registered")
 	}
 	main.impl = &componentImpl{component: main}
 
+	// Initialize client side of the mTLS protocol.
+	if (info.SelfCert == nil) != (info.SelfKey == nil) {
+		return nil, fmt.Errorf(
+			"EnvelopeInfo.{SelfCert, SelfKey} must both be either empty or non-empty")
+	}
+	if info.SelfCert != nil {
+		cert, err := tls.X509KeyPair(info.SelfCert, info.SelfKey)
+		if err != nil {
+			return nil, fmt.Errorf("invalid cert/key pair in EnvelopeInfo: %w", err)
+		}
+		w.selfCert = &cert
+		for cname, c := range w.componentsByName {
+			cname := cname
+			c.clientTLS = &tls.Config{
+				Certificates:       []tls.Certificate{*w.selfCert},
+				InsecureSkipVerify: true, // ok when VerifyPeerCertificate present
```

Comment:
If this is set to `false`, the Go TLS implementation will attempt to verify the server hostname. This means that (1) the hostname field must be set inside the server certificate, and (2) we need to know what hostname to expect. (1) isn't desirable because there are multiple fields other than hostname that may be set inside the server certificate (e.g., DNS name, other arbitrary extensions). In particular, GKE certificates don't set the hostname field. (2) would require new API.